### ISSUE TITLE
Doctest for symbolic_probability gives a warning

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#20314]
```
$ bin/doctest sympy/stats/symbolic_probability.py
============================================================================== test process starts ===============================================================================
executable:         /Users/enojb/current/sympy/38venv/bin/python  (3.8.1-final-0) [CPython]
architecture:       64-bit
cache:              yes
ground types:       gmpy 2.0.8
numpy:              1.19.2
hash randomization: on (PYTHONHASHSEED=3396562498)

sympy/stats/symbolic_probability.py[6] ../Users/enojb/current/sympy/sympy/sympy/stats/rv.py:775: UserWarning: 
Since version 1.7, using `evaluate=False` returns `Expectation`
object. If you want unevaluated Integral/Sum use `E(expr, condition,
evaluate=False).rewrite(Integral)`
  warnings.warn(filldedent(message))
....                                                                                                                                 [OK]

=================================================================== tests finished: 6 passed, in 4.50 seconds ====================================================================
```
What is this warning for?

There shouldn't be a warning for valid code. Otherwise if the code is invalid then it should be removed from the docstring.

@czgdp1807 @Smit-create

**Comment by czgdp1807** [comment]
```python
>>> cexpr.evaluate_integral()
0
```
The above test is giving user warning because, `evaluate=False` in the definition of `_eval_rewrite_as_Integral`.
This is just a user warning for the change in the output type. I don't know if this can be suppressed in the doctests. I don't think removing the warning entirely will be good for the end users and the test is correct so removing that doesn't also isn't a good idea.
IMHO, this is not a really big issue, since, it is intentional and as per plan to notify the end users of the changes in the interface.

**Comment by czgdp1807** [comment]
Is this still a release blocker? May be we can skip the above doc tests if that is the case.

**Comment by czgdp1807** [comment]
> Is this still a release blocker? May be we can skip the above doc tests if that is the case.

ping @oscarbenjamin

**Comment by oscarbenjamin** [comment]
If the test is valid then I don't think that there should be a warning given. Users shouldn't see warnings unless they are doing something that will break in future or something that is always unlikely to work.

**Comment by czgdp1807** [comment]
I see. We can remove the warning then.

**Comment by oscarbenjamin** [comment]
Fixed in #20454
